lambda_functions/review_service/update_review.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import json
+import traceback
+from core_layer import helper
+from core_layer import connection_handler
+from core_layer.handler import user_handler, item_handler, review_handler, review_answer_handler
+from . import notifications
+
+
+def update_review(event, context, is_test=False, session=None):
+    """Updates an existing review
+
+    Args:
+        event: AWS API Gateway event
+        context ([type]): [description]
+        is_test (bool, optional): Whether the function is executed in test mode. Defaults to False.
+        session (optional): An sql alchemy session. Defaults to None.
+
+    Returns
+    ------
+    - Status code 200, 400, 404 or 500 (Created)
+    - The updated review
+    """
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+
+    helper.log_method_initiated("Create Review", event, logger)
+
+    if session == None:
+        session = connection_handler.get_db_session(False, None)
+
+    try:
+        user_id = helper.cognito_id_from_event(event)
+        user = user_handler.get_user_by_id(user_id, is_test, session)
+
+        body = json.loads(event['body']) if isinstance(
+            event['body'], str) else event['body']
+
+        if 'id' not in body:
+            response = {
+                "statusCode": 400,
+                "body": "Bad request. Please provide a review id"
+            }
+            response_cors = helper.set_cors(response, event, is_test)
+            return response_cors
+        else:
+            review = review_handler.get_review_by_id(
+                body['id'], is_test, session)
+
+        if review.user_id != user.id:
+            response = {
+                "statusCode": 403,
+                "body": "Forbidden. You are not allowed to access other users reviews."
+            }
+            response_cors = helper.set_cors(response, event, is_test)
+            return response_cors
+        # If review is set closed
+        if 'status' in body and body['status'] == 'closed':
```

Comment:
I couldn't find the place where the Review is set into the closed state. Is this functionality not implemented yet?

Reply:
It is in review_handler.close_review (line 211).
![image](https://user-images.githubusercontent.com/25669997/113847078-62e59100-9797-11eb-9c9b-36cd0e0a4f1d.png)

Reply:
Thank you. But the close_review function is only called from line 59 (if the Review response status is already "closed"). 

if 'status' in body and body['status'] == 'closed':
            review = review_handler.close_review(review, is_test, session)
            
Who sets the "status" "closed"?

Reply:
This comes from the frontend when the user confirms (and therefore) closes their review.

Reply:
line 58 checks for the request body not for the existing review object.

Reply:
Now I understand. Thank you.